File: vectorDB/scripts/check_record_count.py
#!/usr/bin/env python3
"""
Check actual record count in trademark_embeddings table
"""

import os
import sys
import psycopg2
from pathlib import Path

# Load environment variables from .env.local
def load_env_file(env_path: Path) -> None:
    """Load environment variables from .env.local file"""
    if not env_path.exists():
        return

    with open(env_path, 'r') as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#') and '=' in line:
                key, value = line.split('=', 1)
                # Remove quotes if present
                value = value.strip().strip('"').strip("'")
                os.environ[key] = value

# Load environment variables
project_root = Path(__file__).parent.parent.parent
env_path = project_root / '.env.local'
load_env_file(env_path)

vectordb_env = project_root / 'vectorDB' / '.env.local'
load_env_file(vectordb_env)

# Get connection string
conn_string = os.getenv('POSTGRES_URL_NON_POOLING')

if not conn_string:
    SUPABASE_URL = os.getenv('NEXT_PUBLIC_SUPABASE_URL')
    DB_PASSWORD = os.getenv('POSTGRES_PASSWORD')
    SERVICE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

    if not SUPABASE_URL:
        print("❌ Error: Missing NEXT_PUBLIC_SUPABASE_URL")
        sys.exit(1)

    password = DB_PASSWORD if DB_PASSWORD else SERVICE_KEY
    project_id = SUPABASE_URL.replace('https://', '').replace('.supabase.co', '')

    conn_string = (
        f"postgres://postgres.{project_id}:{password}"
        f"@aws-0-ap-northeast-2.pooler.supabase.com:5432/postgres"
        f"?sslmode=require"
    )

print("🔗 Connecting to Supabase PostgreSQL...")
print()

try:
    conn = psycopg2.connect(conn_string, connect_timeout=30)
    cur = conn.cursor()

    print("✅ Connected successfully")
    print()

    # Get total count
    print("📊 Checking trademark_embeddings table...")
    cur.execute("SELECT COUNT(*) FROM trademark_embeddings;")
    total_count = cur.fetchone()[0]

    print(f"   Total records: {total_count:,}")
    print()

    # Check if specific Starbucks trademarks exist
    starbucks_numbers = ['4520120004900', '4020160015394']

    print("🔍 Checking for specific Starbucks trademarks...")
    for trademark_num in starbucks_numbers:
        cur.execute(
            "SELECT trademark_number FROM trademark_embeddings WHERE trademark_number = %s;",
            (trademark_num,)
        )
        result = cur.fetchone()

        if result:
            print(f"   ✅ Found: {trademark_num}")
        else:
            print(f"   ❌ NOT FOUND: {trademark_num}")

    print()

    # Get sample records
    print("📝 Sample records (first 5):")
    cur.execute("""
        SELECT trademark_number,
               array_length(embedding::float[], 1) as embedding_dim
        FROM trademark_embeddings
        LIMIT 5;
    """)

    for row in cur.fetchall():
        print(f"   {row[0]} - dimension: {row[1]}")

    print()

    # Check embedding dimensions distribution
    print("📏 Checking embedding dimensions...")
    cur.execute("""
        SELECT array_length(embedding::float[], 1) as dim, COUNT(*) as count
        FROM trademark_embeddings
        GROUP BY dim
        ORDER BY count DESC;
    """)

    for row in cur.fetchall():
        print(f"   Dimension {row[0]}: {row[1]:,} records")

    print()

    cur.close()
    conn.close()

    print("=" * 70)
    print("✅ Check Complete")
    print("=" * 70)

except psycopg2.Error as e:
    print(f"❌ Database Error: {e}")
    sys.exit(1)
except Exception as e:
    print(f"❌ Unexpected Error: {e}")
    sys.exit(1)

File: vectorDB/scripts/check-record-count.ts
/**
 * Check actual record count in trademark_embeddings table
 */

import { createClient } from '@supabase/supabase-js'
import * as dotenv from 'dotenv'

// Load environment variables
dotenv.config({ path: '.env.local' })

const SUPABASE_URL = process.env.NEXT_PUBLIC_SUPABASE_URL
const SERVICE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY

if (!SUPABASE_URL || !SERVICE_KEY) {
  console.error('❌ Error: Missing required environment variables')
  console.error('Required: NEXT_PUBLIC_SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY')
  process.exit(1)
}

const supabase = createClient(SUPABASE_URL, SERVICE_KEY, {
  auth: {
    persistSession: false,
    autoRefreshToken: false
  }
})

async function checkRecordCount() {
  console.log('🔗 Connecting to Supabase...')
  console.log(`   URL: ${SUPABASE_URL}`)
  console.log()

  try {
    // Get total count
    console.log('📊 Checking trademark_embeddings table...')
    const { count: totalCount, error: countError } = await supabase
      .from('trademark_embeddings')
      .select('*', { count: 'exact', head: true })

    if (countError) {
      throw countError
    }

    console.log(`   Total records: ${totalCount?.toLocaleString() || 0}`)
    console.log()

    // Check if specific Starbucks trademarks exist
    const starbucksNumbers = ['4520120004900', '4020160015394']

    console.log('🔍 Checking for specific Starbucks trademarks...')
    for (const trademarkNum of starbucksNumbers) {
      const { data, error } = await supabase
        .from('trademark_embeddings')
        .select('trademark_number')
        .eq('trademark_number', trademarkNum)
        .single()

      if (error && error.code !== 'PGRST116') {
        console.warn(`   ⚠️ Error checking ${trademarkNum}:`, error.message)
      } else if (data) {
        console.log(`   ✅ Found: ${trademarkNum}`)
      } else {
        console.log(`   ❌ NOT FOUND: ${trademarkNum}`)
      }
    }

    console.log()

    // Get sample records
    console.log('📝 Sample records (first 10):')
    const { data: samples, error: sampleError } = await supabase
      .from('trademark_embeddings')
      .select('trademark_number')
      .limit(10)

    if (sampleError) {
      throw sampleError
    }

    samples?.forEach((row: any) => {
      console.log(`   ${row.trademark_number}`)
    })

    console.log()

    // Get range of trademark numbers
    console.log('📊 Trademark number range:')
    const { data: firstRecord } = await supabase
      .from('trademark_embeddings')
      .select('trademark_number')
      .order('trademark_number', { ascending: true })
      .limit(1)
      .single()

    const { data: lastRecord } = await supabase
      .from('trademark_embeddings')
      .select('trademark_number')
      .order('trademark_number', { ascending: false })
      .limit(1)
      .single()

    if (firstRecord && lastRecord) {
      console.log(`   First: ${firstRecord.trademark_number}`)
      console.log(`   Last: ${lastRecord.trademark_number}`)
    }

    console.log()
    console.log('='.repeat(70))
    console.log('✅ Check Complete')
    console.log('='.repeat(70))
    console.log()

    if (totalCount && totalCount < 100000) {
      console.log('⚠️  WARNING: Database has fewer than expected records!')
      console.log(`   Expected: ~690,079 records`)
      console.log(`   Actual: ${totalCount.toLocaleString()} records`)
      console.log()
      console.log('💡 This may indicate:')
      console.log('   1. Migration is incomplete')
      console.log('   2. Data was not fully imported')
      console.log('   3. Some records were filtered during import')
      console.log()
    }

  } catch (error) {
    console.error('❌ Error:', error)
    process.exit(1)
  }
}

checkRecordCount()
